### PR TITLE
fix: preserve geographic attribution and alert lifecycle

### PR DESCRIPTION
**Pull request by koala73**
## Summary

Country names no longer become invented country codes: Israel resolves to IL through the existing bundled name catalog before geometry loads, while unknown names remain XX. Cached named signals therefore do not depend on a later replay. Sanctions attribution also tries the country name when its code is unrecognized. Convergence alerts use a verified polygon hit or leave countries unknown, so Berlin and Paris stay separate while nearby events can still merge by distance. All alerts older than 24 hours are removed regardless of insertion or refresh order, keeping the retained list and high-alert count aligned.

Failed or invalid country-geometry loads can retry on the next call. Concurrent callers still share one request, successful base geometry remains cached, and optional boundary overrides still work. Failed indexing also clears partial country mappings.

The repair keeps existing service interfaces and the `XX` unknown-signal convention. A separate attribution/store API was considered but would require unrelated caller migrations. Verified same-country merging remains unchanged for country-level alerts.

## Validation

- Reproduced all reported defects before applying the fixes, plus empty sanctions cluster keys, partial indexes exposed after indexing failure, cached named-signal attribution before geometry, and sanctions name fallback.
- 13 focused Node tests passed: geometry failure/recovery/concurrency, optional overrides, map focus, and signal retention.
- 5 Vitest tests passed: unknown names before/after preload, sanctions/fire attribution, Berlin/Paris separation with and without geometry, nearby merging, mixed-order expiration, refreshed entries, and list/count consistency.
- Pre-push gates also passed: all 1,163 DOM tests and 277 edge/module-isolation tests.
- `npm run typecheck`, `npm run typecheck:dom-tests`, `npm run lint:boundaries`, and `git diff --check` passed.
- Sequential `ce-code-review` persona passes completed; both validated findings were repaired and retested. No independent model review was run under the repository's sequential mapping.
- Inspected desktop (1280px) and mobile (390px) screenshots of the real Strategic Risk panel. Clicking Refresh leaves France and Germany as two rows and reports two high alerts after removing a 25-hour-old Tokyo fixture. Risk-score API and optional override responses were controlled; no page errors. This is component/browser proof, not full dashboard or production-source acceptance.

## Screenshots

Controlled fixture, tested production code at `1b501e078019cd9ce153daf5e90541bc0539eb86`; subsequent changes complete a test fixture and correct signal-name ingestion; the rendered convergence/retention code in these screenshots is unchanged. Uploaded images follow.

## Post-Deploy Monitoring & Validation

After an authorized deployment, the release owner should check the first dashboard session and the next daily refresh: country rows must match event locations, and expired rows must not inflate the High Alerts metric. Search browser logs for `[country-geometry] Failed to load countries.geojson`; after connectivity recovers, a later load should restore geometry. Persistent wrong attribution or failure to recover is a rollback trigger. Deployment and production checks are not part of this PR run.

---

[Compound Engineering](https://github.com/EveryInc/compound-engineering-plugin)
Codex

Desktop controlled fixture at 1b501e078: separate France and Germany alerts, expired Tokyo removed, high-alert count two

Mobile 390px controlled fixture at 1b501e078: same two alert rows and count after Refresh

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved geographic alert attribution using precise location data, keeping nearby locations distinct.
  * Corrected alert cleanup so all alerts older than 24 hours are removed.
  * Improved country identification for sanctions and other location-based data, with unknown locations clearly marked.
  * Added reliable recovery when geographic data loading fails, allowing later attempts to succeed.
  * Prevented duplicate nearby alerts while ensuring refreshed alerts remain current.

* **Tests**
  * Added coverage for geographic loading recovery, country attribution, alert refreshes, and expiration behavior.

### CODE REVIEW

File: src/services/signal-aggregator.ts [hunk 1]
```diff
@@ -141,7 +142,7 @@ class SignalAggregator {
   ingestOutages(outages: InternetOutage[]): void {
     this.clearSignalType('internet_outage');
     for (const o of outages) {
-      const code = normalizeCountryCode(o.country);
+      const code = normalizeCountryCode(o.country) || this.coordsToCountry(o.lat, o.lon);
```
[the remark this answers is not in the document]
Comment (koala73) [reply src/services/signal-aggregator.ts:152]: > this fallback stores named outages as `XX` because both `nameToCountryCode()` and the coordinate lookup are still empty

Not addressing: named outages no longer wait on geometry. `normalizeCountryCode` consults `nameToCountryCode` and `shared/country-names.json` first, then ISO. `tests/dom/geographic-alert-lifecycle.test.mts` asserts Israel/`il` to IL, UK to GB before preload, and Neverland to XX. `refreshGeometryDependentCountryData` still replays only GPS jamming, which has no country hint. Re-ingesting cached outages after geometry would not change named-country attribution.

**Comment by koala73** [comment]
> Dispatch `wm:intelligence-updated` after all alert mutations.

No change. Removals and replacements already update `getRecentAlerts()`. The findings badge also re-reads on its 180s visible-tab poll. Emitting the event from every `updateAlerts()` path is residual polish, not a correctness hole in this PR.

> Reject payloads with no usable polygon geometry before setting `loadedGeoJson`.

No change. Production `countries.geojson` is a full FeatureCollection. An empty-but-typed payload would skip coordinate lookups without retry; that residual stays out of this PR.

> Stage override changes before publishing them.

No change. Override fetch already fails soft and keeps the validated base index. Staging mid-throw mutations is a pre-existing residual, not introduced by the recovery path.

> Docstring coverage is 7.69% which is insufficient.

No change. This repo does not gate on an 80% docstring threshold.
